packages/@aws-cdk/pipelines/lib/legacy/stage.ts

```diff
@@ -225,6 +290,61 @@ export class CdkStage extends CoreConstruct {
     return stripPrefix(s, `${this.stageName}-`);
   }
 
+  /**
+   * Add a security check before the prepare/deploy actions of an CDK stage.
+   * The security check consists of two actions:
+   *  - CodeBuild Action to check for security changes in a stage
+   *  - Manual Approval Action that is auto approved via a Lambda if no security changes detected
+   */
+  private addSecurityCheck(appStageName: string, options?: BaseStageOptions) {
+    const { cdkDiffProject } = this.getApplicationSecurityCheck();
+    const notificationTopic: sns.ITopic | undefined = options?.securityNotificationTopic ?? this.securityNotificationTopic;
+    notificationTopic?.grantPublish(cdkDiffProject);
+
+    const approveActionName = `${appStageName}ManualApproval`;
+    const diffAction = new CodeBuildAction({
+      runOrder: this.nextSequentialRunOrder(),
+      actionName: `${appStageName}SecurityCheck`,
+      input: this.cloudAssemblyArtifact,
+      project: cdkDiffProject,
+      variablesNamespace: `${appStageName}SecurityCheck`,
+      environmentVariables: {
+        STACK_NAME: {
+          value: this.pipelineStage.pipeline.stack.stackName,
+          type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+        },
+        STAGE_NAME: {
+          value: this.stageName,
+          type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+        },
+        ACTION_NAME: {
+          value: approveActionName,
+          type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+        },
+        ...notificationTopic ? {
+          NOTIFICATION_ARN: {
+            value: notificationTopic.topicArn,
+            type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+          },
+          NOTIFICATION_SUBJECT: {
+            value: `Confirm Broadening IAM Permissions in ${appStageName}`,
+            type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+          },
+        } : {},
+      },
+    });
+
+    const approve = new cpactions.ManualApprovalAction({
+      actionName: approveActionName,
+      runOrder: this.nextSequentialRunOrder(),
+      additionalInformation: `#{${appStageName}SecurityCheck.MESSAGE}`,
+      // Boot strap cicd2 us-west-1 --trust
+      externalEntityLink: `https://#{${appStageName}SecurityCheck.LINK}`,
```

Comment:
I saw you saying in the other thing the `https` was added by the manual approval action, but I didn't know WE were the ones making that choice. Is this necessary?

Reply:
LMAO that's embarrassing... i guess this was a change i made early on that i just forgot about.. I think it's better if we remove it and keep it to the script itself